fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -4160,7 +4164,50 @@ public void alterTableProperties(Database db, OlapTable table, Map<String, Strin
                                 ImmutableMap.of(key, propertiesToPersist.get(key)));
                 GlobalStateMgr.getCurrentState().getEditLog().logAlterTableProperties(info);
             }
+            if (propertiesToPersist.containsKey(PropertyAnalyzer.PROPERTIES_DATACACHE_ENABLE)) {
+                String dataCacheEnable = propertiesToPersist.get(PropertyAnalyzer.PROPERTIES_DATACACHE_ENABLE);
+                boolean isEnable = Boolean.parseBoolean(dataCacheEnable);
 
+                Locker locker = new Locker();
+                locker.lockDatabase(db, LockType.WRITE);
+                try {
+                    if (table instanceof LakeTable) {
+                        LakeTable lakeTable = (LakeTable) table;
+                        Collection<Partition> partitions = table.getPartitions();
+                        List<Partition> partitionsList = new ArrayList<>(partitions);
+                        GlobalStateMgr.getCurrentState().getStarOSAgent().updateShardGroup(partitionsList, isEnable);
+                    } else {
+                        throw new DdlException("Only support alter datacache.enable for lake tables");
```

Comment:
fix the error message. e.g. "property 'datacache.enable' is only supported in shared-data mode"

Reply:
OK